tests/mountingOptions/global.stubs.spec.ts

```diff
@@ -853,4 +852,105 @@ describe('mounting options: stubs', () => {
 
     expect(stub.vm).toBe(stubAfterSecondRender.vm)
   })
+
+  describe('directives', () => {
+    const MyDirective: Directive = {
+      beforeMount(el: Element) {
+        el.classList.add('DirectiveAdded')
+      }
+    }
+
+    const MyDirectiveStub: Directive = {
+      beforeMount(el: Element) {
+        el.classList.add('DirectiveStubAdded')
+      }
+    }
+
+    it('stubs directive on root component', () => {
+      const Component = {
+        template: '<div v-my-directive>text</div>',
+        directives: { MyDirective }
+      }
+
+      const wrapper = mount(Component, {
+        global: {
+          stubs: {
+            vMyDirective: MyDirectiveStub
+          }
+        }
+      })
+
+      expect(wrapper.classes()).toContain('DirectiveStubAdded')
+    })
+
+    it('stubs directive as noop', () => {
+      const SomeDirective = () => {
+        throw new Error('I will blow up!')
+      }
+
+      const Component = {
+        template: '<div v-some-directive>text</div>',
+        directives: { SomeDirective }
+      }
+
+      const wrapper = mount(Component, {
+        global: {
+          stubs: {
+            vSomeDirective: true
+          }
+        }
+      })
+
+      expect(wrapper.html()).toBe('<div>text</div>')
+    })
+
+    it('stubs directive on child component', () => {
+      const Component = {
+        template: '<div v-my-directive>text</div>',
+        directives: { MyDirective }
+      }
+
+      const RootComponent = {
+        components: { ChildComponent: Component },
+        template: '<div><child-component class="child" /></div>'
+      }
+
+      const wrapper = mount(RootComponent, {
+        global: {
+          stubs: {
+            vMyDirective: MyDirectiveStub
+          }
+        }
+      })
+
+      expect(wrapper.find('.child').classes()).toContain('DirectiveStubAdded')
+    })
+
+    it('stubs directive on root component with script setup', () => {
+      const Component = {
+        setup() {
+          // @ts-ignore-error (directive used by script setup)
+          const vMyDirective = MyDirective
+        },
+        template: '<div v-my-directive>text</div>',
+        directives: { MyDirective }
+      }
```

Comment:
naive question: shouldn't that use a "real" `script setup` SFC like other tests do? That would seem more bulletproof and close to reality.

Reply:
:white_check_mark: Why not, I have no strong opinion on this one

Reply:
:bomb: Actually this one demonstrated that directives stubbing won't work with `<script setup>`. I will probably go ahead and create an issue for `<script setup>` compiler (I believe we could easily do that there), but for now I just capture current state of the things

Reply:
So this test should be updated to reflect that it does not currently work right?
And ideally, reference the issue you'll create for the compiler, so we can fix it when possible.

Reply:
@cexbrayat nope. `{ setup() ... { ... }}` will still work, `<script setup>` will not. I've added link to https://github.com/vuejs/core/issues/6887 with my proposal to docs to reflect that